### PR TITLE
Update libtorch pinn-pde example

### PR DESCRIPTION
**Pull request by cissieAB**
Make the example executable on both gpu and cpu.
Run successfully on farm A100/T4/TitanRTX GPU and CPU.

```
bash-4.2$ ./install/bin/phasm-example-pinn-pdesolver 
####### A cpp torch example with PINN heat equation. #######

Using CUDA device 0. Training on GPU.
  CUDA device name: NVIDIA A100 80GB PCIe
  CUDA compute capacity: 8.0
  LibTorch version: 1.12.0

y_train sizes: [260, 1]
y_train.device().type(): cuda
y_train.requires_grad(): 0
X_train sizes: [260, 2]
X_train.device().type(): cuda
X_train.requires_grad(): 0
X sizes: [4225, 2]
X.device().type(): cuda
X.requires_grad(): 1


Training started...
  iter=100, loss=9.391394, loss.device().type()=cuda
  iter=200, loss=7.421347, loss.device().type()=cuda
  iter=300, loss=5.533513, loss.device().type()=cuda
  iter=400, loss=0.4877853, loss.device().type()=cuda
  iter=500, loss=0.2155329, loss.device().type()=cuda
  iter=600, loss=0.1304334, loss.device().type()=cuda
  iter=700, loss=0.1099093, loss.device().type()=cuda
  iter=800, loss=0.1007289, loss.device().type()=cuda
  iter=900, loss=0.09455136, loss.device().type()=cuda
  iter=1000, loss=0.08953549, loss.device().type()=cuda

Training stopped.
Final iter=1001, loss=2.763065e-05, loss.device().type()=cuda
bash-4.2$ 
```

### Notes:
- Needs to manually update the libtorch symbolic link to switch between CPU and GPU.
- [ ] `examples/magnetic_field_map` is commented out because it has compiling problems, see issue #4

### CODE REVIEW

File: examples/pinn_pde_solver/CMakeLists.txt [hunk 1]
```diff
@@ -7,5 +7,5 @@ target_include_directories(phasm-example-pinn-pdesolver PRIVATE
 #        ${CMAKE_CURRENT_SOURCE_DIR}/../../surrogate/include
 #        ${CMAKE_CURRENT_SOURCE_DIR}/../../memtrace/include
         )
-target_link_libraries(phasm-example-pinn-pdesolver ${TORCH_LIBRARIES} phasm-surrogate)
```
Comment (nathanwbrei) [inline examples/pinn_pde_solver/CMakeLists.txt:10]: What failed when you tried to link this? I'm attempting to put together a new Docker container where all PHASM components build together (minus CUDA), and I'm going to definitely need your help for the CUDA version.

File: examples/pinn_pde_solver/pinn_diffusion_pde_solver.cpp [hunk 2]
```diff
@@ -39,7 +40,7 @@ const int ADAM_STEPS = 1000;
 const int MAX_STEPS = 5000;
 // criteria for stop training.
 // When loss is at 1.x~e-5, it will stop degrading even the training continues
-const float TARGET_LOSS = 2e-5;
```
Comment (nathanwbrei) [inline examples/pinn_pde_solver/pinn_diffusion_pde_solver.cpp:42]: Is this change for a performance improvement? Or does it simply not terminate if TARGET_LOSS is set too low?

File: examples/pinn_pde_solver/CMakeLists.txt (hunk 1 above)
Comment (cissieAB) [reply examples/pinn_pde_solver/CMakeLists.txt:10]: Nothing failed I guess... Just want to clear some code.

File: examples/pinn_pde_solver/pinn_diffusion_pde_solver.cpp (hunk 2 above)
Comment (cissieAB) [reply examples/pinn_pde_solver/pinn_diffusion_pde_solver.cpp:42]: It's the latter one. Even set to 2e-5, sometimes it would not stop.

**Pull request merged**